packages/kolibri-common/composables/useBaseSearch.js

```diff
@@ -0,0 +1,484 @@
+  const libraryCategories = {};
+
+  const availablePaths = {};
+
+  (categories || []).map(key => {
+    const paths = key.split('.');
+    let path = '';
+    for (const path_segment of paths) {
+      path = path === '' ? path_segment : path + '.' + path_segment;
+      availablePaths[path] = true;
+    }
+  });
+  // Create a nested object representing the hierarchy of categories
+  for (const value of Object.values(Categories)
+    // Sort by the length of the key path to deal with
+    // shorter key paths first.
+    .sort((a, b) => a.length - b.length)) {
+    // Split the value into the paths so we can build the object
+    // down the path to create the nested representation
+    const ids = value.split('.');
+    // Start with an empty path
+    let path = '';
+    // Start with the global object
+    let nested = libraryCategories;
+    for (const fragment of ids) {
+      // Add the fragment to create the path we examine
+      path += fragment;
+      // Check to see if this path is one of the paths
+      // that is available on this device
+      if (availablePaths[path]) {
+        // Lookup the human readable key for this path
+        const nestedKey = CategoriesLookup[path];
+        // Check if we have already represented this in the object
+        if (!nested[nestedKey]) {
+          // If not, add an object representing this category
+          nested[nestedKey] = {
+            // The value is the whole path to this point, so the value
+            // of the key.
+            value: path,
+            // Nested is an object that contains any subsidiary categories
+            nested: {},
+          };
+        }
+        // For the next stage of the loop the relevant object to edit is
+        // the nested object under this key.
+        nested = nested[nestedKey].nested;
+        // Add '.' to path so when we next append to the path,
+        // it is properly '.' separated.
+        path += '.';
+      } else {
+        break;
+      }
+    }
+  }
+  return libraryCategories;
+}
+
+export const searchKeys = [
+  'learning_activities',
+  'categories',
+  'learner_needs',
+  'channels',
+  'accessibility_labels',
+  'languages',
+  'grade_levels',
+];
+
+// TBD #12517 - Will be injected in subsequent work.
+const fetchContentNodeProgress = Promise.resolve({});
+
+export default function useBaseSearch(descendant, store, router) {
+  // Get store and router references from the curent instance
+  // but allow them to be passed in to allow for dependency
+  // injection, primarily for tests.
+  store = store || getCurrentInstance().proxy.$store;
+  router = router || getCurrentInstance().proxy.$router;
+  const route = computed(() => store.state.route);
+
+  const searchResultsLoading = ref(false);
+  const moreLoading = ref(false);
+  const _results = ref([]);
+  const more = ref(null);
+  const labels = ref(null);
+
+  // TODO: Kolibri #12517 - Will be injected in subsequent work
+  const baseurl = '';
```

Comment:
Ah, sorry, the baseurl falsy checking is actually strictly checking for `undefined` so setting this to:
```
const baseurl = undefined;
```

should allow all the tests to pass as expected.